witchcraft/internal/middleware/endpoint_five_hundreds.go

```diff
@@ -0,0 +1,178 @@
+// Copyright (c) 2025 Palantir Technologies. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package middleware
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"slices"
+	"sync"
+	"time"
+
+	"github.com/palantir/witchcraft-go-health/conjure/witchcraft/api/health"
+	"github.com/palantir/witchcraft-go-server/v2/wrouter"
+)
+
+const (
+	endpointFiveHundredsCheckType              = "ENDPOINT_FIVE_HUNDREDS"
+	endpointFiveHundredsWindowCount            = 5
+	endpointFiveHundredsWindowSize             = 2 * time.Minute
+	endpointFiveHundredsMessage                = "There have been HTTP 500s returned by endpoints in every 2 minutes rolling period in the last 10 minutes. This indicates a non-transient error."
```

Comment:
```suggestion
	endpointFiveHundredsMessage                = "There have been HTTP 500s returned by endpoints in every 2 minute rolling period in the last 10 minutes. This indicates a non-transient error."
```

Reply:
I think this is fine since there's not a way to do this without making the message a non-constant, but will note that the message content here is semantically tied to the value of the `endpointFiveHundredsWindowSize` constant but this isn't enforced (if that constant is changed, message will no longer be accurate)

Reply:
Clarified with Brad that this message matches the Java implementation exactly, so keeping as-is